### ISSUE TITLE
mlflow pyfunc serve doesn't correctly execute in the correct conda environment

### ISSUE THREAD

**Issue by frederickaczynski** [mlflow/mlflow#1208]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: Yes, I provided the files below in "Exact command to reproduce".
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 18.04
- **MLflow installed from (source or binary)**: binary
- **MLflow version (run ``mlflow --version``)**: `mlflow, version 0.9.0`
- **Exact command to reproduce**:

Create a folder with the following files:

_MLproject_

```yaml
name: Test project
conda_env: conda.yml
entry_points:
    main:
        command: python train.py
```

_conda.yml_

```yaml
name: test-env
dependencies:
  - pip:
    - loguru
    - mlflow
```

_train.py_

```python
from loguru import logger
import mlflow.pyfunc

logger.debug("Training")

mlflow.log_metric("test", 4)

mlflow.pyfunc.log_model(
    "model",
    "predict",
    code_path=["predict.py"],
    conda_env="conda.yml"
)
```

_predict.py_

```python
from loguru import logger

class MockModel:
    def predict(self):
        logger.debug("Predict")
        return "predictiono"

def _load_pyfunc(datapath):
    logger.debug("Load predict")
    return MockModel()
```

Inside this newly created folder, run `mlflow run .`. It should complete successfuly.

Then, run `mlflow pyfunc serve -m model -r <run_id>`. It should throw an error because it can't find the module `loguru`.

### Describe the problem

The predict script can't find the module `loguru`. By looking at the logs, you can see that `mlflow` create the conda environment and re-execute the command inside this environment by doing: `source /home/fkaczynski/miniconda3/bin/activate mlflow-70df6080b570305b994f0bed58438eadd65de7bd && /usr/local/bin/mlflow pyfunc serve -m model -r <run-id> --no-conda`. You can see that the mlflow binary used is `/usr/local/bin/mlflow`, which is not the mlflow binary that is installed inside the conda environment.

If I execute manually in my terminal `source /home/fkaczynski/miniconda3/bin/activate mlflow-70df6080b570305b994f0bed58438eadd65de7bd && mlflow pyfunc serve -m model -r <run-id> --no-conda`, it works correctly.

### Source code / logs

When running `mlflow run .`:

```
2019/05/07 12:11:20 INFO mlflow.projects: === Created directory /tmp/tmp6pc54qdf for downloading remote URIs passed to arguments of type 'path' ===
2019/05/07 12:11:20 INFO mlflow.projects: === Running command 'source /home/fkaczynski/miniconda3/bin/activate mlflow-70df6080b570305b994f0bed58438eadd65de7bd && python train.py' in run with ID 'f94b4eb806544e8ea03a72c754ca7877' === 
2019-05-07 12:11:21.521 | DEBUG    | __main__:<module>:4 - Training
2019/05/07 12:11:21 INFO mlflow.projects: === Run (ID 'f94b4eb806544e8ea03a72c754ca7877') succeeded ===

```

When running `mlflow pyfunc run -m model -r f94b4eb806544e8ea03a72c754ca7877`:

```
2019/05/07 12:11:44 INFO mlflow.pyfunc.cli: === Running command 'source /home/fkaczynski/miniconda3/bin/activate mlflow-70df6080b570305b994f0bed58438eadd65de7bd && /usr/local/bin/mlflow pyfunc serve -m model -r f94b4eb806544e8ea03a72c754ca7877 --no-conda'
2019/05/07 12:11:45 WARNING mlflow.pyfunc: The version of Python that the model was saved in, `Python 3.7.3`, differs from the version of Python that is currently running, `Python 3.6.7`, and may be incompatible
Traceback (most recent call last):
  File "/usr/local/bin/mlflow", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/mlflow/pyfunc/cli.py", line 71, in serve
    app = scoring_server.init(load_pyfunc(model_path))
  File "/usr/local/lib/python3.6/dist-packages/mlflow/pyfunc/__init__.py", line 288, in load_pyfunc
    return importlib.import_module(conf[MAIN])._load_pyfunc(data_path)
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmp/tmpijimfjo7/model/code/predict.py", line 1, in <module>
    from loguru import logger
ModuleNotFoundError: No module named 'loguru'
```

When running  `source /home/fkaczynski/miniconda3/bin/activate mlflow-70df6080b570305b994f0bed58438eadd65de7bd && /usr/local/bin/mlflow pyfunc serve -m model -r f94b4eb806544e8ea03a72c754ca7877 --no-conda`:

```
2019-05-07 12:12:23.561 | DEBUG    | predict:_load_pyfunc:9 - Load predict
 * Serving Flask app "mlflow.pyfunc.scoring_server" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
```

**Comment by wilmeragsgh** [comment]
Hello @frederickaczynski !
I had a similar issue importing packages and finally got it working importing the packages withing the class definition, in your case, it would be something like:

```python
class MockModel:
    from loguru import logger

    def predict(self):
        MockModel.logger.debug("Predict")
        return "predictiono"

def _load_pyfunc(datapath):
    MockModel.logger.debug("Load predict")
    return MockModel()
```

I hope it works for you too!

**Comment by dbczumar** [comment]
@frederickaczynski The behavior you've described was fixed in MLflow 1.0. Can you try reproducing the behavior on the latest version of MLflow, which is now MLflow 1.1.0? I am going to close the issue for now; please feel free to reopen it if your issue persists.
